Source/JavaScriptCore/wasm/WasmB3IRGenerator.cpp

```diff
@@ -3561,8 +3561,14 @@ auto B3IRGenerator::addThrow(unsigned exceptionIndex, Vector<ExpressionType>& ar
     patch->effects.terminal = true;
     patch->append(instanceValue(), ValueRep::reg(GPRInfo::argumentGPR0));
     patch->append(framePointer(), ValueRep::reg(GPRInfo::argumentGPR1));
-    for (unsigned i = 0; i < args.size(); ++i)
-        patch->append(get(args[i]), ValueRep::stackArgument(i * sizeof(EncodedJSValue)));
+    for (unsigned i = 0; i < args.size(); ++i) {
+        // Note: SIMD values can appear here, but should never be read at runtime because this will throw an un-catchable TypeError instead.
+        // Nonetheless, they may clobber important things if they aren't treated as doubles.
+        auto arg = get(args[i]);
+        if (args[i]->type().isVector())
+            arg = constant(Double, 0);
+        patch->append(arg, ValueRep::stackArgument(i * sizeof(EncodedJSValue)));
+    }
```

Comment:
Do we need the same thing in Air IR generator / Wasm LLInt?

Reply:
Llint never sees SIMD functions, and we do have this in air